subsys/bluetooth/host/Kconfig

```diff
@@ -987,15 +987,17 @@ config BT_ECC
 	help
 	  This option adds support for ECDH HCI commands.
 
-config BT_TINYCRYPT_ECC
-	bool "Emulate ECDH in the Host using TinyCrypt library"
-	select TINYCRYPT
-	select TINYCRYPT_ECC_DH
+config BT_HCI_ECC
```

Comment:
This name "BT_HCI_ECC" is not good at saying what it does. The HCI ECC command exits and may be available whether or not this config is enabled. Any ideas for a better name?

Reply:
Agree. Ideally the `hci_ecc.c` file should be removed. There is no point for the host to go through an emulated hci layer for the ecdh commands.

We already have `CONFIG_BT_ECC` which deals with hci commands, so cant really use that. What about mirroring the crypto configs (`BT_CTLR_CRYPTO`/`BT_HOST_CRYPTO`) and the existing controller ecdh config (`BT_CTLR_ECDH`) with `BT_HOST_ECDH`?

Reply:
"BT_HOST_ECDH" would be short for "Bluetooth Host Elliptic Curve Diffie-Hellman". This is also not describing what this config does.

And it's not really in the Host either. It's used with HCI-raw as well. It's a kind of middleware. Maybe `BT_HCI_ECC_MIDDLEWARE`, `BT_HCI_ECC_INTERCEPT` or `BT_HCI_ECC_EMULATION`?

Reply:
I was thinking we wanted to remove this hci emulation and just have normal function calls in the host

Reply:
I agree, Removing the HCI emulation would be the best.

Reply:
More suggestions for the name: `BT_HCI_ECC_COMPATIBILITY_LAYER` or `BT_HCI_ECC_POLYFILL`.

Reply:
I wanted to avoid mentioning "hci" in the name since that is an implementation detail in the host that we want to remove. What it does is add ecdh support to the host.

Reply:
Thanks for the suggestions! So far I'm still addressing issues with CI, that's why you still see `CONFIG_BT_HCI_ECC` around. Once I'm done with the CI, I will fix the name :)

> Removing the HCI emulation would be the best.

As for this, I think it would be clearer if I do that in a separate PR. Wdyt?

Reply:
> As for this, I think it would be clearer if I do that in a separate PR. Wdyt?

Yes, that can be done later since it is independent of the PSA transition.

Reply:
> I wanted to avoid mentioning "hci" in the name since that is an implementation detail in the host that we want to remove. What it does is add ecdh support to the host.

This a valid perspective, except this config is also used with `CONFIG_BT_HCI_RAW` which we should keep distinct from the Host. Maybe we can say the emulation is not at "HCI" but "`bt_send`". So, `BT_SEND_ECC_EMULATION`?

Reply:
@jhedberg What do you think of the name? And should it even be available for HCI raw?

Reply:
> > I wanted to avoid mentioning "hci" in the name since that is an implementation detail in the host that we want to remove. What it does is add ecdh support to the host.
> 
> This a valid perspective, except this config is also used with `CONFIG_BT_HCI_RAW` which we should keep distinct from the Host. Maybe we can say the emulation is not at "HCI" but "`bt_send`". So, `BT_SEND_ECC_EMULATION`?

HCI raw kind of breaks what I was thinking, with removing the hci commands in the host. If we want to support the ecdh commands in hci raw when the controller does not support them we will have to keep the hci emulation and the hooks in `bt_send`.

Reply:
Code is now updated to `BT_SEND_ECC_EMULATION`. I'm happy with that.